lib/new_relic/agent/instrumentation/mongo.rb

```diff
@@ -16,7 +16,44 @@
 
   executes do
     NewRelic::Agent.logger.info 'Installing Mongo instrumentation'
-    install_mongo_instrumentation
+    if NewRelic::Agent::Datastores::Mongo.is_monitoring_enabled?
+      install_mongo_command_subscriber
+    else
+      install_mongo_instrumentation
+    end
+  end
+
+  class CommandSubscriber
+
+    def started(event)
+      operations[event.operation_id] = event
```

Comment:
Is there a reason that we'll need to hold onto the events between `started` and the finishing callbacks? Seems like there's a potential problem if, for whatever reason, the agent failed to see `succeeded` or `failed` for a command it had seen starting.

If all the necessary information is on the `event` at the end, could we just use it from there?